drivers/serial/Kconfig.bflb

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2021-2024 ATL Electronics
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config UART_BFLB
+	bool "Bouffalo Lab serial driver"
+	depends on DT_HAS_BFLB_BL_UART_ENABLED
+	depends on SOC_FAMILY_BFLB
+	select SERIAL_HAS_DRIVER
+	select SERIAL_SUPPORT_INTERRUPT
```

Comment:
Drivers that require pinmuxing should select `PINCTRL`.

Reply:
It is always defined at SoC folder. BTW, I fixed the sequence to add on the pinctrl commit.

Reply:
> It is always defined at SoC folder. BTW, I fixed the sequence to add on the pinctrl commit.

Then it shouldn't - PINCTRL should be specifically (with some exceptions) be enabled by the individual drivers that need them. There's an ongoing effort for that e.g. here: https://github.com/zephyrproject-rtos/zephyr/issues/78619

Reply:
Fixed!

Side note. I think not having GPIO/PINCTRL at SoC makes no sense. Until today, I could not see a real device without using GPIO and PINCTRL.